src/network/network_udp.cpp

```diff
@@ -171,29 +172,10 @@ void ServerNetworkUDPSocketHandler::Receive_CLIENT_FIND_SERVER(Packet *p, Networ
 	}
 
 	NetworkGameInfo ngi;
-
-	/* Update some game_info */
-	ngi.clients_on     = _network_game_info.clients_on;
-	ngi.start_date     = ConvertYMDToDate(_settings_game.game_creation.starting_year, 0, 1);
-
-	ngi.use_password   = !StrEmpty(_settings_client.network.server_password);
-	ngi.clients_max    = _settings_client.network.max_clients;
-	ngi.companies_on   = (byte)Company::GetNumItems();
-	ngi.companies_max  = _settings_client.network.max_companies;
-	ngi.spectators_on  = NetworkSpectatorCount();
-	ngi.spectators_max = _settings_client.network.max_spectators;
-	ngi.game_date      = _date;
-	ngi.map_width      = MapSizeX();
-	ngi.map_height     = MapSizeY();
-	ngi.map_set        = _settings_game.game_creation.landscape;
-	ngi.dedicated      = _network_dedicated;
-	ngi.grfconfig      = _grfconfig;
-
-	strecpy(ngi.server_name, _settings_client.network.server_name, lastof(ngi.server_name));
-	strecpy(ngi.server_revision, GetNetworkRevisionString(), lastof(ngi.server_revision));
+	FillNetworkGameInfo(ngi);
 
 	Packet packet(PACKET_UDP_SERVER_RESPONSE);
-	this->SendNetworkGameInfo(&packet, &ngi);
+	SendNetworkGameInfo(&packet, &ngi);
```

Comment:
The name is somewhat misleading, yes it was in the past too. It's not actually sending the GameInfo, it's adding it to the packet. Though it does somewhat conform to the functions in Packet, so I'm in a bit of doubt whether it is an issue or not.

Reply:
I agree. The naming comes from these pairs:

```
void ReceiveGRFIdentifier(Packet *p, GRFIdentifier *grf);
void SendGRFIdentifier(Packet *p, const GRFIdentifier *grf);

void ReceiveNetworkGameInfo(Packet *p, NetworkGameInfo *info);
void SendNetworkGameInfo(Packet *p, const NetworkGameInfo *info);
```

Neither are really sending. I could prefix them with `Prepare` or postfix with `Packet` or something, but either they should all change or none, in my opinion. But neither the prefix or postfix sounds good to me .. so I am still looking for a good name :D

Reply:
`Receive` -> `Deserialize` and `Send` -> `Serialize`?